src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsworkload/integration/listener/WorkloadCalculationListenerTests.kt

```diff
@@ -0,0 +1,57 @@
+package uk.gov.justice.digital.hmpps.hmppsworkload.integration.listener
+
+import com.amazonaws.services.sns.model.MessageAttributeValue
+import com.amazonaws.services.sns.model.PublishRequest
+import org.junit.jupiter.api.Assertions
+import org.junit.jupiter.api.Test
+import uk.gov.justice.digital.hmpps.hmppsworkload.domain.CaseType
+import uk.gov.justice.digital.hmpps.hmppsworkload.domain.Tier
+import uk.gov.justice.digital.hmpps.hmppsworkload.domain.event.PersonReference
+import uk.gov.justice.digital.hmpps.hmppsworkload.domain.event.PersonReferenceType
+import uk.gov.justice.digital.hmpps.hmppsworkload.integration.IntegrationTestBase
+import uk.gov.justice.digital.hmpps.hmppsworkload.jpa.entity.CaseDetailsEntity
+import uk.gov.justice.digital.hmpps.hmppsworkload.jpa.entity.PersonManagerEntity
+import uk.gov.justice.digital.hmpps.hmppsworkload.jpa.entity.WorkloadCalculationEntity
+import uk.gov.justice.digital.hmpps.hmppsworkload.listener.AdditionalInformation
+import uk.gov.justice.digital.hmpps.hmppsworkload.listener.WorkloadCalculationEvent
+import java.math.BigDecimal
+import java.math.BigInteger
+import java.time.LocalDateTime
+
+class WorkloadCalculationListenerTests : IntegrationTestBase() {
+
+  @Test
+  fun `calculate workload when staff available hours have changed`() {
```

Comment:
It might be worth adding a negative test, e.g. when the community API call fails